### PR TITLE
Log httperrors and httpservererrors in sentry.

### PR DESCRIPTION
**Pull request by sheetaluk**
Fixes https://github.com/hypothesis/product-backlog/issues/751

### CODE REVIEW

**Comment by robertknight** [comment]
Hi Sheetal - The original issue from @lyzadanger is missing some details about context. I suggest we sync up with her when she is back.

Doing the best I can without that data, what I would expect is:

- If an _expected_ error happens (for example an API is called with invalid parameters, triggering a 4xx response), we wouldn't report that to Sentry.
- If an _unexpected_ error happens (for example due to a bug in the LMS app or missing validation of a parameter, triggering a 5xx response), this should get reported to Sentry. Does that not already happen anyway though?

**Comment by lyzadanger** [comment]
The original issue is not from me…just to clarify

File: lms/views/error.py [hunk 1]
```diff
@@ -26,9 +26,8 @@ def __init__(self, exc, request):
     def httperror(self):
         self.request.response.status_int = self.exc.status_int
         # If code raises an HTTPError or HTTPServerError we assume this was
-        # deliberately raised and:
-        # 1. Show the user an error page including specific error message
-        # 2. _Do not_ report the error to Sentry
+        # deliberately raised and show the user an error page including specific error message
+        self.request.raven.captureException()
```
Comment (robertknight) [inline lms/views/error.py]: Can you provide some explanation for why we're making this change? The comment that was removed says that if code explicitly raises `HTTPError` or `HTTPServerError`, this implies that it was intentional, therefore not a bug and therefore not reported to Sentry. This makes sense.

Presumably there is some context in which the current behaviour is not desirable. What is that?

**Comment by seanh** [comment]
I'm rewriting this part of the code

**Pull request closed without merging**